pkg/test/exporter_collector_test.go

```diff
@@ -207,16 +210,15 @@ func testExporterToCollector(address net.Addr, isMultipleRecord bool, t *testing
 			klog.Fatalf("Got error when sending record: %v", err)
 		}
 		export.CloseConnToCollector() // Close exporting process
-		time.Sleep(time.Second)
-		cp.Stop() // Close collecting process
 	}()
-	go func() {
-		for message := range cp.GetMsgChan() {
-			messages = append(messages, message)
+
+	for message := range cp.GetMsgChan() {
+		messages = append(messages, message)
+		if len(messages) == 2 {
+			break
 		}
```

Comment:
Nit: We could close the msgChan is export go routine after L212 (export.CloseConnToCollector()), instead of having this logic. Feel this is more elegant.

Reply:
makes sense. if we close the msgChan directly after exporter exports message, the messages may not be processed and taken out from the channel. instead I put the channel closing after we get two messages. Thanks!